### PR TITLE
support user setting num_streams=0

### PR DESCRIPTION
**Pull request by wangleis**
### Details:
 - *support user setting num_streams=0*

### Tickets:
 - Closes https://github.com/openvinotoolkit/openvino/issues/20430

### CODE REVIEW

**Review by sunxiaoxia2022 — APPROVED** [review]
I tested it with hello_classification sample. It only spawned one thread by setting parameter stream_num=0. and the result is right.

**Review by xufang-lisa — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
Is it possible to extend the tests to cover such behavior?

**Comment by wangleis** [comment]
> Is it possible to extend the tests to cover such behavior?

Test case updated

File: src/plugins/intel_cpu/tests/functional/behavior/ov_executable_network/properties.cpp [hunk 1]
```diff
@@ -137,6 +137,24 @@ TEST_F(OVClassConfigTestCPU, smoke_CpuExecNetworkCheckModelStreamsHasHigherPrior
     ASSERT_EQ(streams, value);
 }
 
+#if !(defined(OPENVINO_ARCH_ARM) || \
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/tests/functional/behavior/ov_executable_network/properties.cpp:140]: Proper approach is not to disable the test at all, but determine resulting value based on CPU arch. That will lead to failed test once ARM behavior is fixed, wo won't forget to upated the test.

File: src/plugins/intel_cpu/tests/functional/behavior/ov_executable_network/properties.cpp [hunk 2]
```diff
@@ -137,6 +137,24 @@ TEST_F(OVClassConfigTestCPU, smoke_CpuExecNetworkCheckModelStreamsHasHigherPrior
...
+      defined(OPENVINO_ARCH_ARM64))  // Will be removed after multiple streams is supported on ARM
+TEST_F(OVClassConfigTestCPU, smoke_CpuExecNetworkCheckModelZeroStreams) {
+    ov::Core ie;
+    int32_t streams = 0;
+    int32_t value = 100;
```
[the remark this answers is not in the document]
Comment (wangleis) [reply src/plugins/intel_cpu/tests/functional/behavior/ov_executable_network/properties.cpp:145]: @dmitry-gorokhov The resulting value is `0`. The initial value `100` here is used to verify whether `ASSERT_NO_THROW(value = compiledModel.get_property(ov::num_streams))` can get `0`

[the remark this answers is not in the document]
Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/tests/functional/behavior/ov_executable_network/properties.cpp:145]: Than it is better to define with `-1` - it is much more expected default value for uninitialized variable, than `100`

[the remark this answers is not in the document]
Comment (wangleis) [reply src/plugins/intel_cpu/tests/functional/behavior/ov_executable_network/properties.cpp]: Updated

File: src/plugins/intel_cpu/tests/functional/behavior/ov_executable_network/properties.cpp (hunk 1 above)
Comment (wangleis) [reply src/plugins/intel_cpu/tests/functional/behavior/ov_executable_network/properties.cpp]: Updated

**Comment by wangleis** [comment]
@dmitry-gorokhov test case is updated. Please take a look.

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**